Source/plrmsg.cpp

```diff
@@ -25,16 +26,27 @@ namespace {
 
 uint8_t plr_msg_slot;
 _plrmsg plr_msgs[PMSG_COUNT];
+int msgs;
 
 /** Maps from player_num to text color, as used in chat messages. */
 const UiFlags TextColorFromPlayerId[MAX_PLRS + 1] = { UiFlags::ColorWhite, UiFlags::ColorWhite, UiFlags::ColorWhite, UiFlags::ColorWhite, UiFlags::ColorWhitegold };
 
+int GetChatLines(int width, char *textPtr)
+{
+	const size_t length = strlen(textPtr);
+	const string_view text { textPtr, length };
+	std::string lineCountString = WordWrapString(text, width);
+	int lines = std::count(lineCountString.begin(), lineCountString.end(), '\n');
+	lines++;
+	return lines;
+}
+
 void PrintChatMessage(const Surface &out, int x, int y, int width, char *textPtr, UiFlags style)
 {
 	const size_t length = strlen(textPtr);
 	std::replace(textPtr, textPtr + length, '\n', ' ');
 	const string_view text { textPtr, length };
-	DrawString(out, WordWrapString(text, width), { { x, y }, { width, 0 } }, style, 1, 18);
+	DrawString(out, WordWrapString(text, width), { { x, y }, { width, 0 } }, style, 1, 15);

```

Comment:
It would probably be best to move `WordWrapString()` up to a point before you need to know the number of lines. That way we only have to scan the text once and you are sure that your count matches what is actually being drawn.

Reply:
Oh and see if you can get that 15 to come from your `lineHeight` constant.